docs/customizing.rst

```diff
@@ -0,0 +1,81 @@
+Customizing Scipp
+=================
+
+Overview
+--------
+
+A number of concepts and components of **scipp** can and should be customized to the needs of higher-level libraries or to a particular use case.
+At this point we support compile-time customization of:
+
+- Available units and unit combinations
+- Dimension labels
+- The underlying container used for sparse data.
+- New or custom types that can be stored as elements in a ``Variable``.
+- The built-in types that can be support in operations and by the ``transform`` algorithms.
+
+Some of these are adaptable more readily than others, which require more in-depth changes.
+
+Units systems and dimension labels
+----------------------------------
+
+It is possible to select and existing system or to add a new one.
+
+Selecting an existing system
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``cmake`` flag ``SCIPP_UNITS_SYSTEM`` is used to select the system of units (and dimension labels).
+Currently the following options are available:
+
+- ``Dummy``.
+  For unit-testing only.
+  This is incomplete and not useful in practice.
+- ``Neutron``.
+  The system used in ``scipp-neutron``.
+  This contains some SI units as well as a unit for "counts" and some particle-physics-related units like ``meV`` and ``meV/c``.
+
+Creating a new system
+~~~~~~~~~~~~~~~~~~~~~
+
+1. Create files following the same pattern as ``units/include/scipp/units/dummy.h`` and ``units/dummy.cpp``.
+   Adapt the namespace name in both header and source and rename the ``SCIPP_UNITS_DUMMY`` macro.
```

Comment:
I think it would be useful here to take the case of creating a new system called `NewSystem` and follow it through all the steps.
So for example, have:

copy the `units/include/scipp/units/dummy.h` and `units/dummy.cpp` files to `units/include/scipp/units/newsystem.h` and `units/newsystem.cpp`, respectively.

rename the macro `SCIPP_UNITS_DUMMY` to `SCIPP_UNITS_NEWSYSTEM`

etc...

This would make it clearer to follow i think. But this can also wait, it's not urgent.